scipy/sparse/linalg/interface.py

```diff
@@ -437,9 +438,18 @@ def __rmatmul__(self, other):
     def __rmul__(self, x):
         if np.isscalar(x):
             return _ScaledLinearOperator(self, x)
-        else:
+        elif isinstance(x, LinearOperator):
             return NotImplemented
 
+        x = np.asarray(x).conj()
+
+        if x.ndim == 1 or x.ndim == 2 and x.shape[0] == 1:
+            return self.rmatvec(x).conj()
+        elif x.ndim == 2:
+            return self.rmatmat(x).conj()
+        else:
+            raise ValueError(f'expected 1-d or 2-d array or matrix, got {x}')
```

Comment:
This also need a test with `with pytest.raises(ValueError, match="expected 1-d or 2-d array")`. This way you ensure you are catching the right error.